src/startServer.js

```diff
@@ -0,0 +1,69 @@
+import cluster from 'cluster';
+import os from 'os';
+
+import { SERVER_LISTEN_ERROR } from '#lib/logger.const';
+
+// dotenv should be called on entry to the application to ensure all `process.env.*` variables are correctly set from '.env'
+const dotenv = require('dotenv');
+
+const DOT_ENV_CONFIG = dotenv.config();
+if (DOT_ENV_CONFIG.error) {
+  throw DOT_ENV_CONFIG.error;
+}
+
+// now `process.env.*` variables are set run the rest of the app
+const http = require('http');
+const nodeLogger = require('#lib/logger.node');
+const app = require('./server').default;
+
+const logger = nodeLogger(__filename);
+const port = process.env.PORT || 7080;
+
+const startApplicationInstance = () => {
+  const server = http.createServer(app);
+  let currentApp = app;
+  server.listen(port, error => {
+    if (error) {
+      logger.error(SERVER_LISTEN_ERROR, {
+        error,
+      });
+    }
+  });
+
+  if (module.hot) {
+    logger.info('✅  Server-side Hot Module Replacement enabled');
+
+    module.hot.accept('./server', () => {
+      logger.info('🔁  Hot Module Replacement reloading `./server`...');
+      server.removeListener('request', currentApp);
+      const newApp = require('./server').default; // eslint-disable-line global-require
+      server.on('request', newApp);
+      currentApp = newApp;
+    });
+  }
+};
+
+const startCluster = () => {
+  const processOnline = worker => logger.info(`Worker ${worker.id} started`);
+  const processExit = (worker, code, signal) => {
+    const exitReason = code ? ` with code ${code}` : ` due to signal ${signal}`;
+
+    if (code !== 0 && !worker.suicide) {
+      cluster.fork();
+    }
+
+    return logger.error(`Worker ${worker.id} died ${exitReason}`);
```

Comment:
could we try & use structured event logging here, similar to line 27? Makes it super easy for us to run queries against

maybe also reword `died` to `terminated`?

Reply:
updated!